src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,14 +1,47 @@
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { filterSlice } from '../../features/filter';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import debounce from 'lodash.debounce';
 
 export const TodoFilter: React.FC = () => {
+  const dispatch = useDispatch();
+
+  const [query, setQuery] = useState<string>('');
+  const [appliedQuery, setAppliedQuery] = useState<string>('');
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const applyQuery = useCallback(debounce(setAppliedQuery, 1000), []);
```

Comment:
The `useCallback` hook should include `setAppliedQuery` in its dependency array to ensure that the latest version of `setAppliedQuery` is used. This helps avoid potential issues with stale closures.